### PR TITLE
Header, Footer and Burger Menu tests added

### PR DESCRIPTION
**Pull request by LiviuAC**

### CODE REVIEW

File: test/helper/inventoryData.js [hunk 1]
```diff
@@ -47,3 +47,10 @@ exports.ProductDescriptions = {
     redShirt: "This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard " +
         "to automate a few tests. Super-soft and comfy ringspun combed cotton."
 }
+
+exports.BurgerMenu = {
+    allItems: 'All Items',
+    about:'About',
+    logout: 'Logout',
+    resetAppState: 'Reset App State'
+}
```
Comment (axboory) [inline test/helper/inventoryData.js]: Rename to BurgerMenuText

File: test/specs/products.js [hunk 2]
```diff
@@ -197,5 +226,62 @@ describe('Products Page Tests', () => {
...
+        it("the 'Twitter' icon should redirect to the 'https://twitter.com/saucelabs' url", async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+            await ProductsPage.twitterIcon.waitForClickable()
+            await ProductsPage.twitterIcon.click()
+            await browser.switchWindow("https://twitter.com/saucelabs")
+
+            expect(await browser.getUrl()).toEqual("https://twitter.com/saucelabs");
+            await browser.closeWindow()
+            await browser.switchWindow('Swag Labs')
```
Comment (axboory) [inline test/specs/products.js]: You can put these in a method called closeWindowAndSwitchBack or something like that.

File: test/specs/products.js [hunk 3]
```diff
@@ -197,5 +226,62 @@ describe('Products Page Tests', () => {
             // console.log('expected prices: ', dictValues)
         });
     })
-})
 
+    describe('Footer Tests', () => {
+
+        it("check Footer UI elements", async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
```
Comment (axboory) [inline test/specs/products.js:223]: use before each

File: test/specs/products.js [hunk 4]
```diff
@@ -197,5 +226,62 @@ describe('Products Page Tests', () => {
...
+
+            expect(await ProductsPage.twitterIcon.isDisplayed()).toBe(true);
+            expect(await ProductsPage.facebookIcon.isDisplayed()).toBe(true);
+            expect(await ProductsPage.linkedInIcon.isDisplayed()).toBe(true);
+            expect(await ProductsPage.copyright.isDisplayed()).toBe(true);
+            expect(await ProductsPage.copyright.getText()).toContain("© 2022 Sauce Labs. All Rights Reserved. Terms of Service | Privacy Policy");
+            expect(await ProductsPage.robotFooter.isDisplayed()).toBe(true);
+            expect(await ProductsPage.robotFooter.getAttribute('src')).toEqual(ImageSource.footerRobot);
+
+            await ProductsPage.logout();
```
Comment (axboory) [inline test/specs/products.js]: use after each

File: test/specs/products.js [hunk 5]
```diff
@@ -197,5 +226,62 @@ describe('Products Page Tests', () => {
...
+            expect(await browser.getUrl()).toEqual("https://www.facebook.com/saucelabs");
+            await browser.closeWindow()
+            await browser.switchWindow('Swag Labs')
+        })
+
+
+        it("the 'LinkedIn' icon should redirect to the 'https://www.linkedin.com/company/sauce-labs/' url",
+            async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
```
Comment (axboory) [inline test/specs/products.js]: why login again if you are not logging out in previous test?

File: test/specs/products.js [hunk 6]
```diff
@@ -126,30 +123,62 @@ describe('Products Page Tests', () => {
...
+            await ProductsPage.btnBurgerMenu.click();
+            await ProductsPage.btnAbout.waitForClickable();
+            await ProductsPage.btnAbout.click();
+
+            expect(await browser.getUrl()).toEqual("https://saucelabs.com/");
+        })
+
+        it("the 'Logout' button should redirect to the Login page", async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
```
Comment (axboory) [inline test/specs/products.js:162]: why login if you did not logout in previous test?

**Review by axboory — CHANGES_REQUESTED** [review]

File: test/specs/products.js [hunk 7]
```diff
@@ -162,40 +191,114 @@ describe('Products Page Tests', () => {
...
+            await ProductsPage.twitterIcon.waitForClickable()
+            await ProductsPage.twitterIcon.click()
+            await browser.switchWindow("https://twitter.com/saucelabs")
+
+            expect(await browser.getUrl()).toEqual("https://twitter.com/saucelabs");
+            await ProductsPage.closeWindowAndSwitchBack()
+        })
+
+        it("the 'Facebook' icon should redirect to the 'https://www.facebook.com/saucelabs' url",
+            async () => {
```
Comment (axboory) [inline test/specs/products.js]: keep the it line on the same line. shorten the it description if.

File: test/specs/products.js [hunk 8]
```diff
@@ -162,40 +191,114 @@ describe('Products Page Tests', () => {
...
+            await ProductsPage.facebookIcon.waitForClickable()
+            await ProductsPage.facebookIcon.click()
+            await browser.switchWindow("https://www.facebook.com/saucelabs")
+
+            expect(await browser.getUrl()).toEqual("https://www.facebook.com/saucelabs");
+            await ProductsPage.closeWindowAndSwitchBack()
+        })
+
+        it("the 'LinkedIn' icon should redirect to the 'https://www.linkedin.com/company/sauce-labs/' url",
+            async () => {
```
Comment (axboory) [inline test/specs/products.js]: same as above

File: test/specs/products.js [hunk 9]
```diff
@@ -162,40 +191,114 @@ describe('Products Page Tests', () => {
...
+                descendingProductsPrices.push(Number(productPrice[i].replace("$", "")));
+            }
+
+            for (let i = 0; i < dictValues.length; i++) {
+                descendingExpectedProductsPrices.push(Number(dictValues[i].replace("$", "")));
+            }
+            descendingExpectedProductsPrices.sort((a, b) => b - a);
+
+            expect(descendingProductsPrices).toEqual(descendingExpectedProductsPrices);
+            await ProductsPage.filterData('nameAscending')
```
Comment (axboory) [inline test/specs/products.js]: why do you use this method here?

File: test/specs/products.js [hunk 10]
```diff
@@ -162,40 +191,114 @@ describe('Products Page Tests', () => {
...
             }
-            sortedProductsPrices.sort((a, b) => a - b);
-            console.log("sorted prices: ", sortedProductsPrices);
+            ascendingExpectedProductsPrices.sort((a, b) => a - b);
 
-            // let dictValues = Object.values(ProductPrices)
-            // // expectedProductPrices.push(price)
-            // console.log('expected prices: ', dictValues)
+            expect(ascendingProductsPrices).toEqual(ascendingExpectedProductsPrices);
+            await ProductsPage.filterData('nameAscending')
```
Comment (axboory) [inline test/specs/products.js]: why do you use this method here?

File: test/specs/products.js [hunk 11]
```diff
@@ -162,40 +191,114 @@ describe('Products Page Tests', () => {
...
+            const dictValues = Object.values(ProductPrices)
+
+            for (let i = 0; i < productPrice.length; i++) {
+                descendingProductsPrices.push(Number(productPrice[i].replace("$", "")));
+            }
+
+            for (let i = 0; i < dictValues.length; i++) {
+                descendingExpectedProductsPrices.push(Number(dictValues[i].replace("$", "")));
+            }
+            descendingExpectedProductsPrices.sort((a, b) => b - a);
```
Comment (axboory) [inline test/specs/products.js]: this code is repeated, put it in a method.

**Review by axboory — CHANGES_REQUESTED** [review]

File: test/specs/products.js (hunk 7 above)
Comment (LiviuAC) [reply test/specs/products.js]: modification made

File: test/specs/products.js (hunk 8 above)
Comment (LiviuAC) [reply test/specs/products.js]: modification made

File: test/specs/products.js (hunk 9 above)
Comment (LiviuAC) [reply test/specs/products.js]: it was used in order to restore the default change. after further investigation, I have observed that it wasn't needed and it has been removed

File: test/specs/products.js (hunk 10 above)
Comment (LiviuAC) [reply test/specs/products.js]: same as above

File: test/specs/products.js (hunk 11 above)
Comment (LiviuAC) [reply test/specs/products.js]: the method "productsPricesNumberArray" has been created in the ProductsPage file

**Review by axboory — APPROVED** [review]

**Pull request merged**